src/include/distributed/rebalancer_placement_separation.h

```diff
@@ -0,0 +1,40 @@
+/*-------------------------------------------------------------------------
+ *
+ * rebalancer_placement_separation.h
+ *	  Routines to determine which worker node should be used to separate
+ *	  a colocated set of shard placements that need separate nodes.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef PLACEMENT_ISOLATION_H
+#define PLACEMENT_ISOLATION_H
+
+#include "postgres.h"
+#include "nodes/pg_list.h"
+#include "utils/hsearch.h"
+
+#include "distributed/metadata_utility.h"
+
+struct RebalancerPlacementSeparationContext;
+typedef struct RebalancerPlacementSeparationContext RebalancerPlacementSeparationContext;
+
+extern RebalancerPlacementSeparationContext * PrepareRebalancerPlacementSeparationContext(
+	List *activeWorkerNodeList,
+	List
+	*
+	activeShardPlacementList,
+	FmgrInfo
+	*
+	shardAllowedOnNodeUDF);
+extern bool RebalancerPlacementSeparationContextPlacementIsAllowedOnWorker(
+	RebalancerPlacementSeparationContext *context,
+	uint64 shardId,
+	uint64
+	placementId,
+	WorkerNode *
+	workerNode);
```

Comment:
unfortunately citus_indent doesn't allow doing any better, any ideas?

Reply:
I was surprised that it would output it in this horrible way. The only way as was able to fix it was by shortening the type and function names. New uncrustify versions didn't help either.

Reply:
I see, funny thing, lets manually make it less horrendous and tell `citus_indent` to leave it allone

```suggestion
/* *INDENT-OFF* */
extern RebalancerPlacementSeparationContext * PrepareRebalancerPlacementSeparationContext(
	List *activeWorkerNodeList,
	List *activeShardPlacementList,
	FmgrInfo shardAllowedOnNodeUDF);
extern bool RebalancerPlacementSeparationContextPlacementIsAllowedOnWorker(
	RebalancerPlacementSeparationContext *context,
	uint64 shardId,
	uint64 placementId,
	WorkerNode *workerNode);
/* *INDENT-ON* */
```